test/modules/core/core/attribute.spec.js

```diff
@@ -120,3 +120,98 @@ test('WebGL#Attribute getValue', t => {
 
   t.end();
 });
+
+// If the vertex shader has more components than the array provides,
+// the extras are given values from the vector (0, 0, 0, 1) for the missing XYZW components.
+// https://www.khronos.org/opengl/wiki/Vertex_Specification#Vertex_format
+test('Attribute#missing component', t => {
+
+  if (fixture.gl.getExtension('STACKGL_resize_drawingbuffer')) {
```

Comment:
Maybe create a helper function `isHeadlessGL` (in this file)?

Then the code here would read much easier as `if (isHeadlessGL(gl))`...